lib/Connector/XiboAudienceReportingConnector.php

```diff
@@ -209,26 +210,41 @@ public function onRegularMaintenance(MaintenanceRegularEvent $event)
                     $entry['campaignStart'] = $adCampaignCache[$parentCampaignId]['start'];
                     $entry['campaignEnd'] = $adCampaignCache[$parentCampaignId]['end'];
                 } else {
-                    $parentCampaign = $this->campaignFactory->getById($parentCampaignId);
-                    if ($parentCampaign->type == 'ad') {
-                        $adCampaignCache[$parentCampaignId]['type'] = $parentCampaign->type;
-                    } else {
-                        $listCampaignCache[$parentCampaignId] = $parentCampaignId;
-                        continue;
-                    }
+                    // Get Campaign
+                    try {
+                        $parentCampaign = $this->campaignFactory->getById($parentCampaignId);
 
-                    if (!empty($parentCampaign->getStartDt()) && !empty($parentCampaign->getEndDt())) {
-                        $adCampaignCache[$parentCampaignId]['start'] =  $parentCampaign->getStartDt()->format(DateFormatHelper::getSystemFormat());
-                        $adCampaignCache[$parentCampaignId]['end'] = $parentCampaign->getEndDt()->format(DateFormatHelper::getSystemFormat());
-                        $entry['campaignStart'] = $adCampaignCache[$parentCampaignId]['start'];
-                        $entry['campaignEnd'] = $adCampaignCache[$parentCampaignId]['end'];
+                        if ($parentCampaign->type == 'ad') {
+                            $adCampaignCache[$parentCampaignId]['type'] = $parentCampaign->type;
+                        } else {
+                            $listCampaignCache[$parentCampaignId] = $parentCampaignId;
+                            continue;
+                        }
+
+                        if (!empty($parentCampaign->getStartDt()) && !empty($parentCampaign->getEndDt())) {
+                            $adCampaignCache[$parentCampaignId]['start'] =  $parentCampaign->getStartDt()->format(DateFormatHelper::getSystemFormat());
+                            $adCampaignCache[$parentCampaignId]['end'] = $parentCampaign->getEndDt()->format(DateFormatHelper::getSystemFormat());
+                            $entry['campaignStart'] = $adCampaignCache[$parentCampaignId]['start'];
+                            $entry['campaignEnd'] = $adCampaignCache[$parentCampaignId]['end'];
+                        }
+                    } catch (\Exception $exception) {
+                        $erroredCampaign[] = $parentCampaignId;
+                        $event->addMessage(__('Error caching ad/list campaign:') . $parentCampaignId .
+                            ' ' . 'StatId' . ' ' . $entry['id'] .
+                            ' ' . $exception->getMessage());
+                        $this->getLogger()->error('onRegularMaintenance: caching ad/list campaign failed for StatId: ' .
+                            $entry['id'] . ' Errored campaign:' .json_encode($erroredCampaign) . ' ' . $exception->getMessage());
```

Comment:
This will output all of the errored campaigns each time a new error happens - so first time there will be 1, second there will be 1 and 2, third there will be 1, 2, and 3.... is that what you wanted?